mlir/test/Dialect/Ptr/layout.mlir

```diff
@@ -4,36 +4,40 @@ module attributes { dlti.dl_spec = #dlti.dl_spec<
   #dlti.dl_entry<!ptr.ptr, #ptr.spec<size = 32, abi = 32, preferred = 64>>,
   #dlti.dl_entry<!ptr.ptr<5>,#ptr.spec<size = 64, abi = 64, preferred = 64>>,
   #dlti.dl_entry<!ptr.ptr<4>, #ptr.spec<size = 32, abi = 64, preferred = 64, index = 24>>,
+  #dlti.dl_entry<"dlti.default_memory_space", 7 : ui64>,
   #dlti.dl_entry<"dlti.alloca_memory_space", 5 : ui64>,
   #dlti.dl_entry<"dlti.global_memory_space", 2 : ui64>,
   #dlti.dl_entry<"dlti.program_memory_space", 3 : ui64>,
   #dlti.dl_entry<"dlti.stack_alignment", 128 : i64>
 >} {
-  // CHECK: @spec
+  // CHECK-LABEL: @spec
   func.func @spec() {
     // CHECK: alignment = 4
     // CHECK: alloca_memory_space = 5
     // CHECK: bitsize = 32
+    // CHECK: default_memory_space = 7
     // CHECK: global_memory_space = 2
     // CHECK: index = 32
     // CHECK: preferred = 8
     // CHECK: program_memory_space = 3
     // CHECK: size = 4
     // CHECK: stack_alignment = 128
     "test.data_layout_query"() : () -> !ptr.ptr
-    // CHECK: alignment = 4
+    // CHECK: alignment = 1
     // CHECK: alloca_memory_space = 5
-    // CHECK: bitsize = 32
+    // CHECK: bitsize = 64
+    // CHECK: default_memory_space = 7
     // CHECK: global_memory_space = 2
-    // CHECK: index = 32
-    // CHECK: preferred = 8
+    // CHECK: index = 64
```

Comment:
I am a bit confused as to why these values change given that just `#dlti.dl_entry<"dlti.default_memory_space", 7 : ui64>` got added to the data layout spec.

Reply:
Is it because the pointer specs are for different memory spaces?

(Apologies, this is a rather low-level setting I haven't worked with before.)

Reply:
They changed because `!ptr.ptr` (ie. ptr with null memory space) used to automatically define the default address space, therefore `!ptr.ptr<3>` corresponded to the `!ptr.ptr` spec. Since `nullptr` no longer defines the default space, it now returns the globally defined spec (which has those values).